run-tests.php

```diff
@@ -680,10 +680,16 @@ function main(): void
 
     putenv("TEST_PHP_EXECUTABLE=$php");
```

Comment:
on Windows I still get:
```
'C:\Program' is not recognized as an internal or external command
```

Reply:
Which test?

Reply:
To be clear: we still have to keep an unescaped version in TEST_PHP_EXECUTABLE because there are tests which treat this as a path, and not as a command line argument. If you write a test you should use TEST_PHP_EXECUTABLE_ESCAPED.

Reply:
every, I belive the problem is described here https://www.php.net/manual/en/function.proc-open.php for the 1st arg

`'bypass_shell' => true` option added to l1280 fixed the problem for me, but the run cmd looks like: `"C:\Program Files\PHP\php.exe"  ... -f "C:\...\forks\php-src\Zend\tests\anon\015.php"  2>&1` and the stderr redirection is then passed like an arg:

```
print_r(array_slice($argv, 1, null, true));
```
in any phpt test prints:
```
(
    [1] => 2>&1
)
```

one shared pipe reader needs to be probably setup before `proc_open` and then `'bypass_shell' => true` can be used, or even better pass the cmd as array (according to the docs no shell is used then)

Reply:
The problem is that I can't test this myself as I don't have Windows. Everything works on Linux with spaces in the path though.
Could you maybe try to fix it and make a PR against my branch? (https://github.com/nielsdos/php-src/tree/fix-10489)

Reply:
Can you adjust `system_with_timeout` in run-tests.php to accept the cmd as an array accepted by `proc_open` as of PHP 7.4?

Ie. no shell redirections (and no env vars before cmd) in cmd string.

I can then test/fix on my local Windows.

Reply:
Changing it to an array or using the shell_bypass will require a lot of rework.
According to the manual, adding extra quotes should work too. Can you try adding extra quotes in the command line around $php at line 2205? If that works I can just add an `if (system is windows) { add quotes; }` quirk.

Reply:
I did some experiments and doubling, tripling the quotes did not help. But with help of Google I have figured out a way to make string cmd working with quotes as an linux incl. the shell redirection.

PR into your branch: https://github.com/nielsdos/php-src/pull/1

Reply:
Your PR seems to have worked. And the Windows CI also works. The failure in Linux x64 is unrelated.